### PR TITLE
[gardening] Use optional binding

### PR DESCRIPTION
**Pull request by ikesyo**

### CODE REVIEW

**Comment by ikesyo** [comment]
@swift-ci Please test

**Comment by alblue** [comment]
These look good to me. What do you think @phausler ? Do we need to update similar things elsewhere e.g. in overlays?

File: TestFoundation/TestStream.swift [hunk 1]
```diff
@@ -55,9 +55,8 @@ class TestStream : XCTestCase {
         let message: NSString = "Hello, playground"
         let messageData: Data  = message.data(using: String.Encoding.utf8.rawValue)!
         //Initialiser with url
-        let testFile = createTestFile("testFile_in.txt", _contents: messageData)
-        if testFile != nil {
-            let url = URL(fileURLWithPath: testFile!)
+        if let testFile = createTestFile("testFile_in.txt", _contents: messageData) {
```
Comment (ianpartridge) [inline TestFoundation/TestStream.swift]: While these are all improvements, I think perhaps we could do better:

```swift
guard let testFile = createTestFile("testFile_in.txt", _contents: messageData)  else {
   XCTFail("Unable to create temp file")
   return
}
...
```

`guard` is great for reducing the indentation and putting failure code near the test.

**Comment by ikesyo** [comment]
Submitted https://github.com/apple/swift/pull/13314 for some overlays.

**Comment by phausler** [comment]
yea, we should keep the overlays and swift-corelibs-foundation in sync. the change looks reasonable to me (however I didnt really get a chance to go over it in detail but it seems rather mechanical and isomorphic)

**Comment by ikesyo** [comment]
@swift-ci test and merge

**Comment by ikesyo** [comment]
@swift-ci Please test and merge

**Comment by ikesyo** [comment]
@swift-ci Please test and merge

**Comment by ikesyo** [comment]
@swift-ci test and merge

**Comment by ianpartridge** [comment]
@swift-ci test and merge

**Pull request merged**